aips/aip-x.md

```diff
@@ -0,0 +1,247 @@
+
+This has a couple key benefits for users and developers:
+
+- It pushes sophistication and cognitive overhead from users / clients to more specialized entities
+- It allows other entities to compete to provide the best solution for a user, often resulting in better execution / prices
+- It can simplify and speed up cross-chain interactions by pushing risk to "relayers" or "fillers" in exchange for a small fee.
+- Intents are inherently secure, since the outcome signed by the user must match the outcome of the solution. However, ensuring that this is possible typically requires an "intent settlement network".
+
+Intent is a particular concept for Aptos because Move has a unique way of encoding on chain assets. With Move, you can represent
+
+### Overview
+In the proposed intent framework, we see a couple of parties involved:
+1. Intent Creator (User): User will need to offer on chain resources they owned subject to a specific unlock condition.
+2. Intent Contract (dApp): A certain dApp will need to provide programmbale condition when the certain asset could be unlocked.
+3. Intent Solver System: A system will need to monitor the all the intents that's available and create transactions if it finds a lucrative intent. Solver will need to understand the Intent Contract to find out the proper solution.
+4. Intent Verification Framework: Aptos Framework will need to make sure resources can only be unlocked to the solution provider when the post condition contract is executed successfully.
+
+A typical user flow of intent should look like the following:
+1. User creates an intent suggesting:
+    - The asset it's going to offer.
+    - The post condition contract when the asset could be released to the solver.
+    - At the beginning we may integrate this creation process to the wallet with a set of known intent contracts. However, supporting 3rd party intent contract remains unknown to me.
+2. Intent needs to be broadcasted to the interested solver.
+    - In our initial implementation, intent creation could be done by executing a transaction. Such transaction can then emit an event to an event stream that solvers can subscribe themselves to.
+3. Intent Solver will maintain a local pool of intents for each type of intent contract and sythesize a path that would meet the exchange contract.
+4. Intent Solver will sign and submit transactions to process the intent.
+    - In our initial implementation, intent resolution is implemented first come first serve based. The first transaction that can resolve the intent will be able to get the resources wrapped in the intent.
+    - Intent resolution will be done by aptos framework code to make sure the solver did meet the condition listed in the intent contract.
+
+Aptos Labs will be responsible for implementing:
+1. A set of intent contracts to ramp up the system.
+2. Framework code for creating and validating intents.
```

Comment:
Framework code for creating and validating intents.
=> 
Library code for creating and validating intents.